cpp/src/parquet/column_chunker.cc

```diff
@@ -0,0 +1,778 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "parquet/column_chunker.h"
+#include <cmath>
+#include <string>
+#include <vector>
+#include "arrow/array.h"
+#include "arrow/util/logging.h"
+#include "parquet/level_conversion.h"
+
+namespace parquet {
+namespace internal {
+
+constexpr uint64_t GEAR_HASH_TABLE[8][256] = {
```

Comment:
We have 8 independent gear hash tables in order to normalize chunk sizes, see more details in the comments at https://github.com/apache/arrow/pull/45360/files#diff-b0e0dc9efa0271723fc8f200c7a54465dd861655788434ee2884d9bac72e83c2R622

Reply:
Here with dictionary encoding disabled:

![newplot-9](https://github.com/user-attachments/assets/df7d7ed8-62cd-4949-9c08-560bdeb08ccc)

Reply:
Here with dictionary encoding disabled and `norm_factor=1` (instead of the default 0):

![newplot-10](https://github.com/user-attachments/assets/092b4d8f-2e8e-4e7c-a609-2d4ac8f4d3d1)

Reply:
`With norm_factor=2`

![newplot-12](https://github.com/user-attachments/assets/3ba62312-fb85-4a2b-b704-dea0ff75d1fc)

Reply:
With `norm_factor=3`

![newplot-13](https://github.com/user-attachments/assets/f68689e6-d050-4410-9a24-5d7c53e97393)

Reply:
With `norm_factor=8`

![newplot-14](https://github.com/user-attachments/assets/0593afaf-502d-4694-be63-b72f249a26de)